<!-- Driver for the standard joystick (/dev/input/js0). -->
<launch>
  <node pkg="joy" exec="joy_node" name="joystick_driver" output="screen">
    <param name="device_id" value="0" />
    <param name="dev" value="/dev/input/js0" />
    <param name="deadzone" value="0.5" />
    <param name="autorepeat_rate" value="20.0" />
    <param name="sticky_buttons" value="false" />
    <param name="coalesce_interval" value="1" />
  </node>

  <!--  smooths inputs from cmd_vel_mux/input/teleop_raw to cmd_vel_mux/input/teleop -->
  <!--include file="$(find turtlebot_teleop)/launch/includes/velocity_smoother.launch.xml"/-->

  <!--- Teleop Joystick, taken from iai_boxy node -->
  <node pkg="teleop_twist_joy" exec="teleop_node" name="turtlebot_teleop_joystick" respawn="true">
    <param name="enable_button" value="4" />
    <param name="axis_linear.x" value="1"/>
    <param name="axis_angular.yaw" value="3"/>
    <param name="scale_linear.x" value="0.4"/>
    <param name="scale_angular.yaw" value="1.2"/>   
    <remap from="/cmd_vel" to="/ps3/cmd_vel"/>

  </node>

    <!-- Throttle laser frequency to 25 Hz -->
  <node pkg="topic_tools" exec="throttle" name="scan_throttle">
    <param name="throttle_type" value="messages"/>
    <param name="rate" value="25.0"/>
    <param name="input_topic" value="/scan"/>
    <param name="output_topic" value="/scan_throttle"/>
    <param name="output_message_type" value="sensor_msgs/msg/LaserScan"/>
  </node>


  <!-- Throttle /cmd_vel (TwistStamped) to 10 Hz -->
  <node pkg="topic_tools" exec="throttle" name="cmd_vel_throttle">
    <param name="throttle_type" value="messages"/>
    <param name="rate" value="10.0"/>  <!-- 10 Hz -->
    <param name="input_topic" value="/ps3/cmd_vel"/>
    <param name="output_topic" value="/ps3/cmd_vel_throttled"/>
    <param name="output_message_type" value="geometry_msgs/msg/Twist"/>
  </node>


</launch>